libmozevent/bus.py

```diff
@@ -105,6 +117,21 @@ async def _get():
 
             return await _get()
 
+    async def restore_redis_messages(self):
+        """
+        Restores currently processed Redis messages
+        Only sequential jobs can be restored this way, parallel jobs will be ignored
```

Comment:
Correct me if I'm wrong: if there are parallel jobs that are each using a different queue, then this will work flawlessly. The problem is only for parallel jobs that use the same queue.